src/proof/trust_node.h

```diff
@@ -126,6 +126,8 @@ class TrustNode
    * provide a proof for this fact.
    */
   Node getProven() const;
+  /** set generator */
+  void setGenerator(ProofGenerator* g);
```

Comment:
I'd prefer keeping TrustNode immutable, you can use `mkReplaceGenTrustNode` instead of doing this.

Reply:
haha I implemented this method exactly because I wanted something like `mkReplaceGenTrustNode` but it seemed overkill. I had not noticed it. I'll change to use it.